compiler/rustc_hir_typeck/src/place_op.rs

```diff
@@ -109,7 +109,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         index_ty: Ty<'tcx>,
         index_expr: &hir::Expr<'_>,
     ) -> Option<(/*index type*/ Ty<'tcx>, /*element type*/ Ty<'tcx>)> {
-        let adjusted_ty = self.structurally_resolve_type(autoderef.span(), autoderef.final_ty());
```

Comment:
this means we no longer eagerly error on infer vars here. surprised we don't have a test for it. needs test + check for infer

Reply:
i think
```rust
fn foo() {
    let x = &Default::default();
    x[1];
    let _: &Vec<()> = x;
}
``` would work